### PR TITLE
add puppet-strings documentation

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -1,9 +1,18 @@
+# @summary module to configure a systemd timer + script to backup postgres, mysql and ldap dataabases
+#
+# @param pigz_cpu_cores number of cpu cores used for backup compression
```
Comment (ekohl) [inline manifests/init.pp:3]: I'm a fan of a newline. So
```suggestion
# @param pigz_cpu_cores
#   number of cpu cores used for backup compression
```

(and repeated elsewhere).

File: manifests/init.pp [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+# @param backuphistory number of days to keep the backup history
+# @param cleanup_empty_backup_dirs enable/disable a post command to cleanup empty directories with $destination. They can occour if the system run out of free diskspace
+#
+# @author Tim Meusel <tim@bastelfreak.de>
 class dbbackup (
-  Integer $pigz_cpu_cores             = $facts['processors']['count'],
+  Integer[1] $pigz_cpu_cores          = $facts['processors']['count'],
   Stdlib::Absolutepath $destination   = '/mnt/dumps',
-  String $interval                    = '1h',
+  String[2] $interval                 = '1h',
```
Comment (alexjfisher) [inline manifests/init.pp:13]: What about `'30m'` or `'12h'`?  Are these not valid?

Comment (alexjfisher) [reply manifests/init.pp:13]: Ignore me.  Need more sleep!

File: manifests/init.pp (hunk 1 above)
Comment (bastelfreak) [reply manifests/init.pp:3]: ok, will do

File: manifests/init.pp (hunk 2 above)
Comment (ekohl) [reply manifests/init.pp:18]: Those are accepted. `2` is the min length, not the exact length. If it was `String[2, 2]` it would be a problem.

**Pull request merged**